usl_models/usl_models/flood_ml/model.py

```diff
@@ -118,12 +173,82 @@ def forward(
 
         # Concatenate and feed into remaining ConvLSTM and TransposeConv layers
         # [B, n, H', W', k'] -> [B, H, W, 1]
-        lstm_input = self.concat([st_cnn_output, geo_cnn_output, temp_input])
+        lstm_input = tf.concat([st_cnn_output, geo_cnn_output, temp_input], axis=-1)
         lstm_output = self.conv_lstm(lstm_input)
         output = self.output_cnn(lstm_output)
 
         return output
 
-    def call(self):
-        """Runs the entire autoregressive model."""
-        raise NotImplementedError
+    def call(self, inputs):
+        """Runs the entire autoregressive model.
+
+        Args:
+            inputs: A dictionary of input tensors.
+                - "geospatial" (required): [H, W, f]
+                - "temporal" (required): [T]
+                - "spatiotemporal" (optional): a single flood map [H, W]
+
+        Returns:
+            A tuple (all_flood_predictions, max_flood_map) of the flood predictions and
+            the max predicted flood depth at each cell over the entire storm duration.
+            The maps are the same shape as the input: (H, W).
+        """
+        # TODO: Add data validation
+        geo_input = inputs.get("geospatial")
+        full_temp_input = data_utils.temporal_window_view(
+            inputs.get("temporal"), self.params["m_rainfall"]
+        )
+        # Note: We assume that, if provided, this input is a *single* flood map.
+        # For convenience, we add this to self.flood_maps to align indexing
+        # between flood maps and rainfall ,i.e., the current flooding conditions
+        # and rainfall at time t are stored at index t along the temporal axis.
+        st_input = inputs.get("spatiotemporal")
+        if st_input is None:
+            st_shape = geo_input.shape[:3] + [1]
+            st_input = tf.zeros(st_shape)
+        self.flood_maps.write(0, st_input)
+
+        # We use 1-indexing for simplicity. Time step t represents the t-th flood
```

Comment:
Is it common to use 1-indexing in the land of modelling? It breaks my tiny computer science brain to use anything besides 0-indexing!

Reply:
This isn't a design choice because of ML practices or anything (I think zero-indexing is probably the default still). I mostly did this for better interpretability. With 1-indexing, time t represents the t-th prediction (in layman terms). It also makes indexing into our tensors a bit more consistent; otherwise the `_update_temporal_inputs` method ends up having some tensors indexed at `t` and some at `t+1`, which I found unnecessarily confusing.